### PR TITLE
PM-10266: Master password guidance screen

### PR DESCRIPTION
**Pull request by shannon-livefront**
## 🎟️ Tracking

[PM-10266](https://bitwarden.atlassian.net/browse/PM-10266)

## 📔 Objective

Create the "Master password guidance" informational screen.

This screen is not currently reachable in the app since we don't have the updated implementation of the "choose your master password" screen with the "what makes a password strong" button on it. To view this screen, you can call `showMasterPasswordGuidance()` from one of the other routes in `AuthCoordinator.navigate(to:context:)`.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-10266]: https://bitwarden.atlassian.net/browse/PM-10266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceProcessor.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /// - Parameter coordinator: The coordinator that handles navigation.
+    ///
+    init(coordinator: AnyCoordinator<AuthRoute, AuthEvent>) {
+        self.coordinator = coordinator
+        super.init(state: ())
+    }
+
+    // MARK: Methods
+
+    override func perform(_: Void) async {}
```
Comment (fedemkr) [inline BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceProcessor.swift:28]: ⛏️ Can this method be removed as it's doing nothing?

File: BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceView.swift [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            VStack(alignment: .leading, spacing: 8) {
+                Text(Localizations.whatMakesAPasswordStrong)
+                    .styleGuide(.title3, weight: .semibold)
+
+                Text(Localizations.strongPasswordDescriptionLong)
+                    .styleGuide(.subheadline)
+            }
+            .frame(maxWidth: .infinity, alignment: .leading)
+            .foregroundStyle(Asset.Colors.textPrimary.swiftUIColor)
+            .padding(.all, 24)
```
Comment (fedemkr) [inline BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceView.swift:50]: ⛏️ No need for `.all` as it's the default.
```suggestion
            .padding(24)
```

File: BitwardenShared/UI/Auth/CompleteRegistration/CompleteRegistrationProcessorTests.swift [hunk 3]
```diff
@@ -675,5 +675,12 @@ class CompleteRegistrationProcessorTests: BitwardenTestCase {
         subject.receive(.toastShown(Toast(text: "example")))
         XCTAssertEqual(subject.state.toast?.text, "example")
     }
+
+    /// `receive(_:)` with `.learnMoreTapped` launches the master password guidance view.
+    @MainActor
+    func test_receive_learnMoreTapped() {
```
Comment (matt-livefront) [inline BitwardenShared/UI/Auth/CompleteRegistration/CompleteRegistrationProcessorTests.swift:681]: ⛏️ Could you alphabetize this test relative to the others?

File: BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceViewTests.swift [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        try button.tap()
+        XCTAssertEqual(processor.dispatchedActions.last, .dismiss)
+    }
+
+    /// Tapping the password generator button dispatches the `.dismiss` action.
+    @MainActor
+    func test_passwordGeneratorButton_tap() throws {
+        let button = try subject.inspect().find(button: Localizations.cancel)
+        try button.tap()
+        XCTAssertEqual(processor.dispatchedActions.last, .dismiss)
```
Comment (matt-livefront) [inline BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceViewTests.swift:45]: ⚠️ This should be testing the generate password button, not cancel.

File: BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceView.swift [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    ///
+    private func bulletPoint(text: String) -> some View {
+        HStack(alignment: .top, spacing: 6) {
+            Text("•")
+
+            Text(text)
+        }
+        .styleGuide(.subheadline)
+        .foregroundStyle(Asset.Colors.textSecondary.swiftUIColor)
+    }
```
[the remark this answers is not in the document]
Comment (fedemkr) [reply BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceView.swift:117]: In the simulator (iPhone 15 Pro iOS 17.0.1), it's not currently changing. Although I see it doesn't change language correctly either. Testing on my device now.

[the remark this answers is not in the document]
Comment (fedemkr) [reply BitwardenShared/UI/Auth/CompleteRegistration/MasterPasswordGuidance/MasterPasswordGuidanceView.swift:117]: Same behavior on Device, it seems that there's an issue on RTL languages. So I think we're good for this PR and we can tackle the RTL languages problem in another PR.

**Review by matt-livefront — APPROVED** [review]
Looks good!

**Pull request merged**